kudo-operator/templates/spark-history-server-deployment.yaml

```diff
@@ -0,0 +1,64 @@
+{{- if .Params.enableHistoryServer }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: history-server
+  namespace: {{ .Namespace }}
+  labels:
+    app.kubernetes.io/name: history-server
+    app.kubernetes.io/instance: {{ .Name }}
+    app.kubernetes.io/version: {{ .Params.operatorVersion }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: history-server
+      app.kubernetes.io/instance: {{ .Name }}
+      app.kubernetes.io/version: {{ .Params.operatorVersion }}
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: history-server
+        app.kubernetes.io/instance: {{ .Name }}
+        app.kubernetes.io/version: {{ .Params.operatorVersion }}
+    spec:
+      {{- if .Params.createOperatorServiceAccount }}
+      serviceAccountName: {{ .Name }}-{{ .Params.operatorServiceAccountName }}
+      {{- else }}
+      serviceAccountName: {{ .Params.operatorServiceAccountName }}
+      {{- end }}
+      containers:
+      - name: history-server
+        image: {{ .Params.operatorImageName }}:{{ .Params.operatorVersion }}
+        imagePullPolicy: {{ .Params.imagePullPolicy }}
+        command: ["/opt/spark/sbin/start-history-server.sh"]
+        env:
+        - name: SPARK_NO_DAEMONIZE
+          value: "true"
+        - name: AWS_ACCESS_KEY_ID
+          valueFrom:
+            secretKeyRef:
+              name: s3-secret
+              key: accessKeyID
+        - name: AWS_SECRET_ACCESS_KEY
+          valueFrom:
+            secretKeyRef:
+              name: s3-secret
+              key: secretAccessKey
+        - name: SPARK_HISTORY_OPTS
```

Comment:
how about adding only this single field to `params.yaml` and let users specify all the options they want? So it can look like that:
```
- name: SPARK_HISTORY_OPTS
  value: "-Dspark.history.fs.logDirectory={{ .Params.historyServerFsLogDirectory }} {{.Params.historyServerOptions }}"
```

Reply:
using secrets is a nice approach though, but I don't think it must be done in the scope of this PR. Most of the AWS users rely on IAM and instance profiles for access management so it doesn't seem like the best option for them. And S3 is not the only supported storage... So let's skip the secrets for now and allow users to provide them via `{{.Params.historyServerOptions }}`